sshd-core/src/main/java/org/apache/sshd/client/channel/PtyCapableChannelSession.java

```diff
@@ -271,4 +302,34 @@ protected void doOpenPty() throws IOException {
 
         sendEnvVariables(session);
     }
+
+    protected byte[] getXCookie() {
+        final Session session = getSession();
+        Object xCookie = ChannelX11.X11_COOKIE.getOrNull(session);
+        Object xCookieHex = ChannelX11.X11_COOKIE_HEX.getOrNull(session);
+        if (xCookie instanceof byte[] && xCookieHex instanceof byte[]) {
+            return (byte[]) xCookieHex;
+        }
+
+        synchronized (session) {
+            xCookieHex = ChannelX11.X11_COOKIE_HEX.getOrNull(session);
+            if (xCookieHex instanceof byte[]) {
+                return (byte[]) xCookieHex;
+            }
+
+            byte[] foo = new byte[16];
```

Comment:
Give variables meaningful names. "foo" and "bar" just won't do. Here "foo" could be named "fakeCookie".